modules/social_features/social_group/modules/social_group_welcome_message/social_group_welcome_message.module

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for social_group_welcome_message module.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\group\Entity\GroupContentInterface;
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function social_group_welcome_message_entity_base_field_info(EntityTypeInterface $entity_type) {
+  $fields = [];
+
+  // Add a 'Welcome message' base fields to all group types.
+  if ($entity_type->id() === 'group') {
+    $fields['private_message_send'] = BaseFieldDefinition::create('boolean')
+      ->setLabel(t('Send a private message people joined'))
+      ->setRevisionable(TRUE)
+      ->setTranslatable(TRUE)
+      ->setDefaultValue(FALSE)
+      ->setDisplayOptions('form', [
+        'type' => 'boolean_checkbox',
+        'settings' => [
+          'display_label' => TRUE,
+        ],
+        'weight' => 99,
+      ])
+      ->setDisplayConfigurable('form', TRUE);
+
+    $fields['private_message_body'] = BaseFieldDefinition::create('text_long')
+      ->setLabel(t('Message'))
+      ->setRevisionable(TRUE)
+      ->setTranslatable(TRUE)
+      ->setDisplayOptions('form', [
+        'type' => 'text_textfield',
+        'weight' => 100,
+      ])
+      ->setDisplayConfigurable('form', TRUE);
+  }
+
+  return $fields;
```

Comment:
Do these need to be two dependant fields? I think we can get away with a single welcome message field here that can be nullable.

If no text is filled in then no message will be sent. If you want the user experience of having a checkbox with a hideable text field then this can be done in a Field Widget instead but does not need to be stored. That also simplifies your form alter logic below.

Reply:
In my opinion better will be to keep these fields as they are because I think it's not good to check text field if it's empty or not and also GM can prepare body of welcome message, exclude checkbox and just store text without implementing base functionality of welcoming message.

But if you really want to adjust this then we can

Reply:
Okay